modules/internal/ChapelLocale.chpl

```diff
@@ -179,10 +179,10 @@ module ChapelLocale {
       :returns: locale number, in the range ``0..numLocales-1``
       :rtype: int
      */
-    proc id : int return chpl_id();  // just the node part
+    proc id : int return chpl_nodeFromLocaleID(__primitive("_wide_get_locale", this));
```

Comment:
@mppf While working on https://github.com/chapel-lang/chapel/pull/11579 I found myself wondering if I could/should use `_wide_get_node` in the network atomic code, which also made me wonder if it would be appropriate here. I'm not very familiar with the differences between` _wide_get_locale` and` _wide_get_node` though so this could be a silly suggestion.

That also led me to https://github.com/chapel-lang/chapel/commit/70aa131bd8, which looks like it might have been an older form of this optimization, but I'm not sure if that optimization still triggers.

Reply:
AFAIK _wide_get_node is there for historical reasons and _wide_get_locale should be used in preference to it in order to make explicit the ability to handle sublocales.